packages/node_modules/@webex/internal-plugin-devices/src/devices.js

```diff
@@ -0,0 +1,795 @@
+     * request. This property gets set via the initial registration process by a
+     * `this.set()` method.
+     *
+     * @type {string}
+     */
+    url: 'string',
+
+    /**
+     * This property stores the device's userId uuid value, which can also be
+     * derived from the device's registerd user's userId retrieved from
+     * the **Hydra** service.
+     *
+     * @type {string}
+     */
+    userId: 'string',
+
+    /**
+     * This property stores whether or not file sharing is enabled
+     *
+     * @type {'BLOCK_BOTH' | 'BLOCK_UPLOAD'}
+     */
+    webFileShareControl: 'string',
+
+    /**
+     * This property stores the current web socket url used by the registered
+     * device.
+     *
+     * @type {string}
+     */
+    webSocketUrl: 'string',
+
+    /**
+     * This property stores the value indicating whether or not white board file
+     * sharing is enabled for the current device.
+     *
+     * @type {'ALLOW' | 'BLOCK'}
+     */
+    whiteboardFileShareControl: 'string'
+  },
+
+  derived: {
+    /**
+     * This property determines if the current device is registered.
+     *
+     * @type {boolean}
+     */
+    registered: {
+      deps: ['url'],
+
+      /**
+       * Checks if the device is registered by validating that the url exists.
+       *
+       * @returns {boolean}
+       */
+      fn() {
+        return !!(this.url);
+      }
+    }
+  },
+
+  /**
+   * Stores timer data as well as other state details.
+   *
+   * @type {Object}
+   */
+  session: {
+
+    /**
+     * This property stores the logout timer object
+     *
+     * @type {any}
+     */
+    logoutTimer: 'any',
+
+    /**
+     * This property stores the date for the last activity the user made
+     * with the current device.
+     *
+     * @type {number}
+     */
+    lastUserActivityDate: 'number',
+
+    /**
+     * This property stores whether or not the reachability check has been
+     * performed to prevent the reachability check from performing its
+     * operation more than once after a successful check.
+     *
+     * @returns {boolean}
+     */
+    isReachabilityChecked: ['boolean', false, false],
+
+    /**
+     * This property stores whether or not the current device is in a meeting
+     * to prevent an unneeded timeout of a meeting due to inactivity.
+     *
+     * @type {boolean}
+     */
+    isInMeeting: 'boolean',
+
+    /**
+     * This property identifies if the device is currently in network to prevent
+     * the `resetLogoutTimer()` method from being called repeatedly once its
+     * known client is connected to the organization's internal network.
+     *
+     * @type {boolean}
+     */
+    isInNetwork: 'boolean'
+  },
+
+  // Event method members.
+
+  /**
+   * Trigger meeting started event for webex instance. Used by web-client team.
+   *
+   * @returns {void}
+   */
+  meetingStarted() {
+    this.webex.trigger('meeting started');
```

Comment:
It would be good to have these events as constants that gets exported as well, that way users can import the event name variable.

See the meetings plugin for reference:
https://github.com/webex/webex-js-sdk/blob/master/packages/node_modules/@webex/plugin-meetings/src/constants.js#L345